### ISSUE TITLE
dbBind() doesn't support vector arguments

### ISSUE THREAD

**Issue by warnes** [r-dbi/odbc#259]
The DBI man page for `dbBind` states
> The elements of the params argument do not need to be scalars, vectors of arbitrary length (including length 0) are supported. For queries, calling `dbFetch()` binding such parameters returns concatenated results, equivalent to binding and fetching for each set of values and connecting via `rbind()`. 

However, this does not appear to be implemented in the `odbc` package (at least for the Microsoft SQL connection):
```
> library(odbc)
> 
> con <- dbConnect(odbc(),
+                  Driver = "/opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.3.so.1.1",
+                  Server = "###.###.###.###",
+                  Database = "XXXXXXXXXX",
+                  UID = "XXXXXXX",
+                  PWD = "XXXXXXX")
> 
> params <- list(ID=c('28262', '28269'))
> 
> sql <- "SELECT DISTINCT location_cd, DBA_nm 
+         FROM dw.vwDim_Customer 
+         WHERE location_cd=?"
> 
> ###
> # Provide params a singleton, twice and join
> ###
> 
> qry <- dbSendQuery(con, sql)
> dbBind(qry, params=params[[1]][1] )
> query_1 <- dbFetch(qry)
> dbClearResult(qry)
> 
> qry <- dbSendQuery(con, sql)
> dbBind(qry, params=params[[1]][2])
> query_2 <- dbFetch(qry)
> dbClearResult(qry)
> 
> rbind(query_1, query_2)
  location_cd                DBA_nm
1       28262 ESTABROOK FORD NISSAN
2       28269 GARY'S PAINT AND BODY
> 
> 
> ###
> # Provide params as a list containing a vector of values
> # this should yield the same results
> ###
> qry <- dbSendQuery(con, sql)
> dbBind(qry, params=params )
> final_query <- dbFetch(qry)
> dbClearResult(qry)
> 
> final_query
  location_cd                DBA_nm
1       28262 ESTABROOK FORD NISSAN
```

<details>
<summary>Session Info</summary>

```r
> devtools::session_info()
─ Session info ────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.5.2 (2018-12-20)
 os       Ubuntu 18.04.2 LTS          
 system   x86_64, linux-gnu           
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Etc/UTC                     
 date     2019-03-02                  

─ Packages ────────────────────────────────────────────────────────────────────────────────────
 package     * version date       lib source                     
 assertthat    0.2.0   2017-04-11 [2] CRAN (R 3.5.0)             
 backports     1.1.3   2018-12-14 [2] CRAN (R 3.5.1)             
 bit           1.1-14  2018-05-29 [2] CRAN (R 3.5.0)             
 bit64         0.9-7   2017-05-08 [2] CRAN (R 3.5.0)             
 blob          1.1.1   2018-03-25 [1] CRAN (R 3.5.1)             
 callr         3.1.1   2018-12-21 [2] CRAN (R 3.5.2)             
 cli           1.0.1   2018-09-25 [2] CRAN (R 3.5.1)             
 crayon        1.3.4   2017-09-16 [2] CRAN (R 3.5.0)             
 DBI           1.0.0   2018-05-02 [1] CRAN (R 3.5.1)             
 desc          1.2.0   2018-05-01 [2] CRAN (R 3.5.0)             
 devtools      2.0.1   2018-10-26 [2] CRAN (R 3.5.1)             
 digest        0.6.18  2018-10-10 [2] CRAN (R 3.5.1)             
 fs            1.2.6   2018-08-23 [2] CRAN (R 3.5.1)             
 glue          1.3.0   2018-07-17 [2] CRAN (R 3.5.1)             
 hms           0.4.2   2018-03-10 [2] CRAN (R 3.5.0)             
 knitr         1.21    2018-12-10 [2] CRAN (R 3.5.1)             
 magrittr      1.5     2014-11-22 [2] CRAN (R 3.5.0)             
 memoise       1.1.0   2017-04-21 [2] CRAN (R 3.5.0)             
 odbc        * 1.1.6   2019-03-01 [1] Github (r-dbi/odbc@555ea0d)
 pkgbuild      1.0.2   2018-10-16 [2] CRAN (R 3.5.1)             
 pkgconfig     2.0.2   2018-08-16 [2] CRAN (R 3.5.1)             
 pkgload       1.0.2   2018-10-29 [2] CRAN (R 3.5.1)             
 prettyunits   1.0.2   2015-07-13 [2] CRAN (R 3.5.0)             
 processx      3.2.1   2018-12-05 [2] CRAN (R 3.5.1)             
 ps            1.3.0   2018-12-21 [2] CRAN (R 3.5.2)             
 R6            2.4.0   2019-02-14 [2] CRAN (R 3.5.2)             
 Rcpp          1.0.0   2018-11-07 [2] CRAN (R 3.5.1)             
 remotes       2.0.2   2018-10-30 [2] CRAN (R 3.5.1)             
 rlang         0.3.1   2019-01-08 [2] CRAN (R 3.5.2)             
 rprojroot     1.2     2017-01-16 [2] CRAN (R 3.5.0)             
 rstudioapi    0.9.0   2019-01-09 [2] CRAN (R 3.5.2)             
 sessioninfo   1.1.1   2018-11-05 [2] CRAN (R 3.5.1)             
 usethis       1.4.0   2018-08-14 [2] CRAN (R 3.5.1)             
 withr         2.1.2   2018-03-15 [2] CRAN (R 3.5.0)             
 xfun          0.5     2019-02-20 [2] CRAN (R 3.5.2)             

[1] /usr/local/lib/R/site-library
[2] /usr/lib/R/site-library
[3] /usr/lib/R/library
```
</details>

**Comment by stefanedwards** [comment]
Do mind you, that you do not have to clear and re-send the query between each value, i.e.

```{R}
qry <- dbSendStatement(con, sql)
dbBind(qry, params=params[[1]][1])
result1 <- dbFetch(qry)
dbBind(qry, params=params[[1]][2])
result2 <- dbFetch(qry)
dbClearResult(qry)
rbind(result1, result2)
```

But I too am troubled by this issue.

**Comment by DavorJ** [comment]
This is a R [`odbc`](https://cran.r-project.org/web/packages/odbc/index.html) package issue, or SQL Server odbc driver issue?

**Comment by jimhester** [comment]
The odbc package was first released before multi-row binding was added to the spec and DBItest spec suite.

Supporting it in the {odbc} package would require rewriting a how results are fetched to account for the possibility of multiple parameter binding, and I (at least) don't have the time or inclination to do so. You can always work around this limitation by performing the queries and row binding yourself as demonstrated above.

This is unlikely to be fixed without an external PR.

**Comment by vkapartzianis** [comment]
I submitted a patch to resolve the issue, see https://github.com/r-dbi/odbc/pull/345

**Comment by jimhester** [comment]
Should now be fixed by #345
